### PR TITLE
[improve](Status) Add new status code`KEY_NOT_FOUND` and `KEY_ALREADY_EXISTS` for merge on write

### PR DESCRIPTION
**Pull request by bobhan1**
## Proposed changes

For merge on write table, when loading data, it will call `lookup_row_key` to find if there exists a row with the same key as the inserted row. The return value of `lookup_row_key` can be one of `Status::NotFound`/`Status::AlreadyExist`/`Status::OK`.  Currently, call to `Status::NotFound`/`Status::AlreadyExist` will lead to printing stacktrace info. However, the number of call to `Status::NotFound`/`Status::AlreadyExist` can be up to **the number of rows inserted times the number segments of the tablet** in the worst case. So it may print lots of useless stacktrace info when loading data. So this PR add two new status code for this scenarios to eliminate printing these useless stacktrace info.

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by bobhan1** [comment]
run buildall

**Review by Yukang-Lian — APPROVED** [review]
LGTM

File: be/src/olap/rowset/segment_v2/segment.cpp [hunk 1]
```diff
@@ -367,14 +367,14 @@ Status Segment::lookup_row_key(const Slice& key, bool with_seq_col, RowLocation*
 
     DCHECK(_pk_index_reader != nullptr);
     if (!_pk_index_reader->check_present(key_without_seq)) {
-        return Status::NotFound("Can't find key in the segment");
+        return Status::Error<ErrorCode::KEY_NOT_FOUND>("Can't find key in the segment");
```
Comment (zhannngchen) [inline be/src/olap/rowset/segment_v2/segment.cpp:370]: Use `ERROR_CTOR(KeyNotFound, KEY_NOT_FOUND)` is more clear?

**Review by liaoxin01 — APPROVED** [review]
LGTM

File: be/src/olap/rowset/segment_v2/segment.cpp (hunk 1 above)
Comment (bobhan1) [reply be/src/olap/rowset/segment_v2/segment.cpp:370]: It seems that `ERROR_CTOR ` is for Errors defined in TStatus. Here, `KEY_NOT_FOUND` and `KEY_ALREADY_EXSITS` are be internal errors

Comment (zhannngchen) [reply be/src/olap/rowset/segment_v2/segment.cpp:370]: got it

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 46.09 seconds
 stream load tsv:          536 seconds loaded 74807831229 Bytes, about 133 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          64 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          28.7 seconds inserted 10000000 Rows, about 348K ops/s
 storage size: 17161880480 Bytes

**Comment by bobhan1** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 46.05 seconds
 stream load tsv:          536 seconds loaded 74807831229 Bytes, about 133 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          64 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          28.9 seconds inserted 10000000 Rows, about 346K ops/s
 storage size: 17162004987 Bytes

**Review by zhannngchen — APPROVED** [review]
LGTM

**Pull request merged**